wg-code-quality/wg-charter.md

```diff
@@ -0,0 +1,64 @@
+# WG Code-Quality Charter
+
+## Scope
+
+The Code-Quality Working Group has two goals:
+
+* (short-term) **focus on the improvement of code quality** and
+* (long-term) **create an inter-SIG process for identifying, agreeing on and addressing code quality issues**
+
+The long-term goal is achieved when the process is implemented and executed as required.
+
+## In scope
+
+The main objective of this effort is to continuously improve the quality of the project codebase.
+
+With this in mind, there is a need to define a clear scope on which areas and through what means this can be accomplished.
+
+* Create and maintain code-quality guidelines, best practices and standards.
+* Assure distribution of knowledge in regard to code quality.
+  * This may include presentations, documentation, lectures, certifications, linters, etc.
+* Enforcement of selected standards.
+  * Through automation and active intervention in code reviews.
+  * Through codebase infrastructure. E.g reusable libraries.
+* Management:
+  * Prioritizing tasks.
+  * Focusing on selected topics.
+* Monitoring and identification of code-quality issues.
+
+The WG will work on facilitating and managing the quality effort until enough tooling, documentation, code structure, reviewer and maintainer awareness is achieved.
+
+## Stakeholders
+
+### Special Interest Groups
+
+* All SIGS claiming code ownership and producing code, currently
+  * SIG Compute,
+  * SIG Network and
+  * SIG Storage
+* sig-ci
+  * automation purposes of code-quality measures
+
+### Roles and responsibilities
+
+#### Chairs [1]
+- @EdDev
+- @iholder101
+
+#### Liaisons
+- @dhiller
+
+## Deliverables
+
+The inter-SIG process definition will be manifested in one or more documents that define the quality effort:
+* what tooling and what documentation is required,
+* what metrics for code structure should be established and how it will be implemented,
```

Comment:
nit:
```suggestion
* what tooling and what documentation is required
* what metrics for code structure should be established and how it will be implemented
```